include/seqan3/io/sequence_file/input.hpp

```diff
@@ -644,28 +642,100 @@ class sequence_file_input
             return;
         }
 
-        assert(!format.valueless_by_exception());
-        std::visit([&] (auto & f)
+        format->read_sequence_record(*secondary_stream, record_buffer, options);
+    }
+
+    /*!\brief An abstract base class to store the selected input format.
+     *
+     * \details
+     *
+     * This abstract base class is used to store the user given input format as a type-erased base class in a
+     * std::unique_ptr. There is exactly one derived type that implements this abstract base class called
+     * seqan3::sequence_file_input::selected_input_format which is a private subclass of
+     * seqan3::sequence_file_input. It is not exposed to the user and allows to hide the implementation detail of
+     * storing a specfic format instance which is first known at runtime.
+     */
+    struct selected_input_format_base
```

Comment:
```suggestion
    struct sequence_format_base
```

or

```suggestion
    struct selected_sequence_format
```
 Other suggestions?

Reply:
I'm not sure, but I think it would be more clear if we would name it a bit more specific.

This base class can only be used within this class and only supports sequence file formats.

Reply:
Well, the intention was to refactor this into a base class in the next steps to get rid of the duplicated code. 
In this case, you would have something more generic like I chose here. But the renaming can also be done later I suppose.